Add backup feature to user module

samdoran opened the pull request:
##### SUMMARY

Add feature to backup shadow file only on platforms where the module modifies the shadow file directly, which are currently SunOS variants.

Fixes #40696 

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
`user.py`

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
2.7
```

samdoran committed:
Add backup option

samdoran committed:
Only backup shadow file when the OS has one

samdoran committed:
Only backup shadow file for SunOS

samdoran committed:
Update docs on backup feature

samdoran committed:
Add changelog fragment

bcoca commented on lib/ansible/modules/system/user.py:
```diff
@@ -194,6 +194,14 @@
         type: bool
         default: 'no'
         version_added: "2.4"
+    backup:
```
i would not even make this an option, specially since it behaves differntly than other 'backup' options, we should just 'do this'

samdoran replied:
Ok. I'll just add a note then and remove the option.

samdoran committed:
Add tests for shadow backup

samdoran committed:
Remove backup option, make it automatic

Remove the option to enable/disable backups and make it automatic. Add note to docs describing this behavior.

Change tests to account for new module behavior.

Change section name in changelog fragment since minor_features is not a valid section.

samdoran commented:
rebuild_merge

samdoran commented:
rebuild_merge
